src/components/todo-list-edit-modal/TodoListEditModal.tsx

```diff
@@ -0,0 +1,188 @@
+import DatePicker from 'react-datepicker';
+import { Category, Color } from '@todo/enums';
+import { TodoItem } from '@todo/store';
+
+const newTodoItem: TodoItem = {
+  title: '',
+  description: '',
+  isImportant: false,
+  isCompleted: false,
+  createdDate: new Date().toDateString(),
+};
+
+type TodoItemProps = {
+  todoItem: TodoItem;
+  isEdit: false;
+  onClose: boolean;
+};
+
+const TodoListEditModal = ({ todoItem, isEdit, onClose }: TodoItemProps) => {
+  if (!isEdit) {
+    todoItem = newTodoItem;
+  }
+
+  return (
+    <>
+      <div className="relative">
+        <div className="absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 top-20">
+          <div
+            className="bg-white shadow-xl p-7 rounded-md border border-slate-300 overflow-hidden"
+            style={{ width: '500px' }}
+          >
+            <button className="absolute text-blue-600 underline" style={{ top: '0', right: '-50px' }} onClick={onClose}>
+              Close
+            </button>
+            <form action="">
+              <div className="mb-4">
+                <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="title">
+                  Title
+                </label>
+                <input
+                  id="title"
+                  name="title"
+                  type="text"
+                  defaultValue={todoItem.title}
+                  placeholder="Enter title"
+                  className="shadow border rounded w-full py-2 px-3 text-gray-700 leading-tight"
+                />
+              </div>
+
+              <div className="mb-4">
+                <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="description">
+                  Description
+                </label>
+                <textarea
+                  id="description"
+                  name="description"
+                  defaultValue={todoItem.description}
+                  placeholder="Description"
+                  className="shadow border rounded w-full py-2 px-3 text-gray-700 leading-tight"
+                />
+              </div>
+
+              <div className="mb-4">
+                <label className="block text-gray-700 text-sm font-bold mb-2" htmlFor="category">
+                  Category
+                </label>
+                <select
+                  id="category"
+                  name="category"
+                  defaultValue={todoItem.category}
+                  className="shadow border rounded w-full py-2 px-3 text-gray-700 leading-tight"
```

Comment:
Where is the `onChnage` attribute?

Reply:
I will implement it later. In this PR just popup for new and edit cases.